runtime/vm/compiler/assembler/assembler_arm.cc

```diff
@@ -1781,13 +1781,16 @@ void Assembler::StoreIntoObject(Register object,
   // store buffer (case 1) or add value to the marking stack (case 2).
   // Compare UntaggedObject::StorePointer.
   Label done;
+#ifndef DART_OPTIMIZE_FOR_SIZE
   if (can_be_smi == kValueCanBeSmi) {
```

Comment:
@mraleph I would like to assess the performance impact if we inline SMI checking again, how do you think?

Reply:
I thought about this a bit more and I think inlining SMI checking fast path is not that helpful. Most of the integer fields should be unboxed now, so the only place where it would help is some large collection of integer values. (e.g. `List<int>`). But developers should not be using such collections to begin with and instead use `dart:typed_data`.

Reply:
make sense.

Reply:
@mraleph thanks for your feedback!